core/storage/question/gae_models.py

```diff
@@ -284,12 +361,19 @@ def get_question_skill_links_equidistributed_by_skill(
         question_count_per_skill = int(
             math.ceil(float(total_question_count) / float(len(skill_ids))))
         question_skill_link_models = []
+
         for skill_id in skill_ids:
-            question_skill_link_models.extend(
-                cls.query(cls.skill_id == skill_id).fetch(
-                    question_count_per_skill
-                )
-            )
+            question_ids = [
+                model.question_id for model in question_skill_link_models]
+
+            query = cls.query(cls.skill_id == skill_id)
+            # Deduplicate if the same question is linked to multiple skills.
+            for existed_question_id in question_ids:
```

Comment:
existed_question_id -> existing_question_id

Reply:
Done.

Reply:
This hasn't changed?

Reply:
Oh sorry, I only changed one of the functions. Now it's done.